prow/plugins/lgtm/lgtm.go

```diff
@@ -334,6 +349,67 @@ func checkAllowedToLGTM(
 	return true, nil
 }
 
+// returns ok = true if the TBR is allowed in this case
+// also returns an errors encountered while handling
+// may comment on the PR
+func checkAllowedToTBR(
+	rc reviewCtx,
+	skipCollaborators bool,
+	gc githubClient,
+	ownersClient repoowners.Interface,
+	log *logrus.Entry,
+) (ok bool, err error) {
+	// get fields from rc
+	commentAuthor := rc.author
+	prAuthor := rc.issueAuthor
+	number := rc.number
+	body := rc.body
+	htmlURL := rc.htmlURL
+	org := rc.repo.Owner.Login
+	repoName := rc.repo.Name
+
+	// only the Author can TBR their PR, anyone else should use LGTM
+	isPRAuthor := commentAuthor == prAuthor
+	if !isPRAuthor {
+		resp := "you cannot TBR someone else's PR."
+		log.Infof("Commenting with \"%s\".", resp)
+		return false, gc.CreateComment(rc.repo.Owner.Login, rc.repo.Name, rc.number, plugins.FormatResponseRaw(rc.body, rc.htmlURL, commentAuthor, resp))
+	}
+
+	// the author must be a collaborator or in owners
+	// if skipCollaborators is not enabled, we can ask github if the author
+	// is a collaborator on the repo
+	if !skipCollaborators {
+		ok, err = gc.IsCollaborator(org, repoName, commentAuthor)
+		msg := ""
+		if err == nil && !ok {
+			msg = fmt.Sprintf(", and only %s/%s repo collaborators may be assigned issues", org, repoName)
+		} else if err != nil {
+			log.WithError(err).Errorf("Failed IsCollaborator(%s, %s, %s)", org, repoName, commentAuthor)
+		}
+		resp := "TBR is restricted to assignees" + msg + "."
+		log.Infof("Reply to assign via /lgtm request with comment: \"%s\"", resp)
+		return false, gc.CreateComment(org, repoName, number, plugins.FormatResponseRaw(body, htmlURL, commentAuthor, resp))
+	}
+	// otherwise we depend on OWNERS files instead to check if the author
+	// is an approver or reviwer of the changed files
+	log.Debugf("Skipping collaborator checks and loading OWNERS for %s/%s#%d", org, repoName, number)
```

Comment:
We might want to add a field to the logger to distinguish this flow from that for `lgtm` as they come from the same plugin, emit identical logging messages and are otherwise going to be impossible to differentiate